drivers/hmis/app/graphql/types/hmis_schema/query_type.rb

```diff
@@ -209,6 +209,21 @@ def static_form_definition(role:)
       Hmis::Form::Definition.order(:id).with_role(role).first!
     end
 
+    field :form_definition_for_json, Types::Forms::FormDefinitionForJsonResult, null: true do
+      argument :input, String, required: true
+    end
+    def form_definition_for_json(input:)
+      json = JSON.parse(input)
+      errors = []
+      ::HmisUtil::JsonForms.new.tap do |builder|
+        builder.validate_definition(json, on_error: ->(err) { errors << err })
+      end
+
+      return { errors: errors, definition: nil } if errors.present?
+
+      return { errors: [], definition: json }
+    end
```

Comment:
This *feels* to me like it should be a Mutation (`validateFormDefinitionJson` or `parseFormDefinitionJson`), since it's not actually querying any data. (It's not mutating any either, though, so I could see an argument either way).

Was there a particular reason you decided to go with a query over a mutation?

The name `form_definition_for_json` feels a bit off to me, I think `parse` or `validate` would be clearer about what it's actually doing.

Reply:
Yeah, it's a bit borderline. I decided to go with a query because it feels like we're asking the backend for information –"Is this JSON a valid form definition? If so, please process it and give me the full definition schema"–rather than telling it to change anything. I think it feels mutation-y because we're asking it to transform something, but that something is just the `input` we passed in, which feels like a valid thing to have a query do as well. Also, since the input is JSON, that makes it feel like a POST request, which typically aligns with mutations, so I think that makes it feel mutation-y as well.

I do think it makes more sense as a query given that it's really just asking for information. Regardless I can make it a mutation instead if you think that's better, there's no mechanical reason it couldn't be a mutation.

I'd be fine with changing the name, I agree it's awkward. Maybe we could go with `parsed_form_definition` if we keep it a query? If we switch to mutation, I think `parse_form_definition` would be fine.